File: backend/SindhDormitory.Domain/Entities/Department.cs
// =============================================================================
// Domain/Entities/Department.cs
// =============================================================================
namespace SindhDormitory.Domain.Entities;

public class Department : BaseEntity
{
    public int    DepartmentId { get; set; }
    public string Name         { get; set; } = string.Empty;
    public string Code         { get; set; } = string.Empty;  // e.g. "CS", "EE"

    // Navigation
    public ICollection<Program>                 Programs               { get; set; } = [];
    public ICollection<UniversityStudentRecord> UniversityStudentRecords { get; set; } = [];
}

File: backend-node/src/entities/Department.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { Program } from './Program';

@Entity('Departments')
export class Department {
  @PrimaryGeneratedColumn({ name: 'DepartmentId' })
  departmentId!: number;

  @Column({ name: 'Name', type: 'varchar', length: 150 })
  name!: string;

  @Column({ name: 'Code', type: 'varchar', length: 20, unique: true })
  code!: string;

  @CreateDateColumn({ name: 'CreatedAt' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'UpdatedAt' })
  updatedAt!: Date;

  @OneToMany(() => Program, (p) => p.department)
  programs!: Program[];
}
